Enhancement/eks adding service ipv4 range

philnichol opened the pull request:
<!--- See what makes a good Pull Request at : https://github.com/terraform-providers/terraform-provider-aws/blob/master/docs/CONTRIBUTING.md --->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

<!--- Thank you for keeping this note for the community --->

<!--- If your PR fully resolves and should automatically close the linked issue, use Closes. Otherwise, use Relates --->
Closes #15505

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):
<!--
If change is not user facing, just write "NONE" in the release-note block below.
-->

```release-note
Added service_ipv4_cidr input to aws_eks_cluster.
```

Output from acceptance testing:

<!--
Replace TestAccXXX with a pattern that matches the tests affected by this PR.

For more information on the `-run` flag, see the `go test` documentation at https://tip.golang.org/cmd/go/#hdr-Testing_flags.
-->
```
$ TF_ACC=1 go test ./aws -v -count 1 -parallel 4 -run=TestAccAWSEksCluster_ -timeout 120m
=== RUN   TestAccAWSEksCluster_basic
=== PAUSE TestAccAWSEksCluster_basic
=== RUN   TestAccAWSEksCluster_EncryptionConfig
=== PAUSE TestAccAWSEksCluster_EncryptionConfig
=== RUN   TestAccAWSEksCluster_Version
=== PAUSE TestAccAWSEksCluster_Version
=== RUN   TestAccAWSEksCluster_Logging
=== PAUSE TestAccAWSEksCluster_Logging
=== RUN   TestAccAWSEksCluster_Tags
=== PAUSE TestAccAWSEksCluster_Tags
=== RUN   TestAccAWSEksCluster_VpcConfig_SecurityGroupIds
=== PAUSE TestAccAWSEksCluster_VpcConfig_SecurityGroupIds
=== RUN   TestAccAWSEksCluster_VpcConfig_EndpointPrivateAccess
=== PAUSE TestAccAWSEksCluster_VpcConfig_EndpointPrivateAccess
=== RUN   TestAccAWSEksCluster_VpcConfig_EndpointPublicAccess
=== PAUSE TestAccAWSEksCluster_VpcConfig_EndpointPublicAccess
=== RUN   TestAccAWSEksCluster_VpcConfig_PublicAccessCidrs
=== PAUSE TestAccAWSEksCluster_VpcConfig_PublicAccessCidrs
=== CONT  TestAccAWSEksCluster_basic
=== CONT  TestAccAWSEksCluster_VpcConfig_SecurityGroupIds
=== CONT  TestAccAWSEksCluster_Logging
=== CONT  TestAccAWSEksCluster_Version
--- PASS: TestAccAWSEksCluster_Logging (944.83s)
=== CONT  TestAccAWSEksCluster_VpcConfig_PublicAccessCidrs
--- PASS: TestAccAWSEksCluster_VpcConfig_SecurityGroupIds (957.92s)
=== CONT  TestAccAWSEksCluster_EncryptionConfig
--- PASS: TestAccAWSEksCluster_basic (992.14s)
=== CONT  TestAccAWSEksCluster_VpcConfig_EndpointPublicAccess
--- PASS: TestAccAWSEksCluster_EncryptionConfig (907.86s)
=== CONT  TestAccAWSEksCluster_VpcConfig_EndpointPrivateAccess
--- PASS: TestAccAWSEksCluster_VpcConfig_PublicAccessCidrs (1211.42s)
=== CONT  TestAccAWSEksCluster_Tags
--- PASS: TestAccAWSEksCluster_Version (2577.52s)
--- PASS: TestAccAWSEksCluster_VpcConfig_EndpointPublicAccess (1640.84s)
--- PASS: TestAccAWSEksCluster_Tags (840.69s)
--- PASS: TestAccAWSEksCluster_VpcConfig_EndpointPrivateAccess (2191.08s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	4058.874s
```
Adds a kubernetes_network_configuration block, currently the only field in the block is service_ipv4_cidr, which sets the IP range allocated to k8s services within a cluster

philnichol commented:
reran acceptance test (with newly added acceptance test and latest commit that sets minItems on list:
```
$ TF_ACC=1 go test ./aws -v -count 1 -parallel 4 -run=TestAccAWSEksCluster_ -timeout 120m
=== RUN   TestAccAWSEksCluster_basic
=== PAUSE TestAccAWSEksCluster_basic
=== RUN   TestAccAWSEksCluster_EncryptionConfig
=== PAUSE TestAccAWSEksCluster_EncryptionConfig
=== RUN   TestAccAWSEksCluster_Version
=== PAUSE TestAccAWSEksCluster_Version
=== RUN   TestAccAWSEksCluster_Logging
=== PAUSE TestAccAWSEksCluster_Logging
=== RUN   TestAccAWSEksCluster_Tags
=== PAUSE TestAccAWSEksCluster_Tags
=== RUN   TestAccAWSEksCluster_VpcConfig_SecurityGroupIds
=== PAUSE TestAccAWSEksCluster_VpcConfig_SecurityGroupIds
=== RUN   TestAccAWSEksCluster_VpcConfig_EndpointPrivateAccess
=== PAUSE TestAccAWSEksCluster_VpcConfig_EndpointPrivateAccess
=== RUN   TestAccAWSEksCluster_VpcConfig_EndpointPublicAccess
=== PAUSE TestAccAWSEksCluster_VpcConfig_EndpointPublicAccess
=== RUN   TestAccAWSEksCluster_VpcConfig_PublicAccessCidrs
=== PAUSE TestAccAWSEksCluster_VpcConfig_PublicAccessCidrs
=== RUN   TestAccAWSEksCluster_NetworkConfig_ServiceIpv4Cidr
=== PAUSE TestAccAWSEksCluster_NetworkConfig_ServiceIpv4Cidr
=== CONT  TestAccAWSEksCluster_basic
=== CONT  TestAccAWSEksCluster_VpcConfig_EndpointPrivateAccess
=== CONT  TestAccAWSEksCluster_NetworkConfig_ServiceIpv4Cidr
=== CONT  TestAccAWSEksCluster_VpcConfig_PublicAccessCidrs
--- PASS: TestAccAWSEksCluster_basic (841.95s)
=== CONT  TestAccAWSEksCluster_VpcConfig_EndpointPublicAccess
--- PASS: TestAccAWSEksCluster_NetworkConfig_ServiceIpv4Cidr (935.96s)
=== CONT  TestAccAWSEksCluster_Logging
--- PASS: TestAccAWSEksCluster_VpcConfig_PublicAccessCidrs (1052.45s)
=== CONT  TestAccAWSEksCluster_VpcConfig_SecurityGroupIds
--- PASS: TestAccAWSEksCluster_Logging (880.39s)
=== CONT  TestAccAWSEksCluster_Tags
--- PASS: TestAccAWSEksCluster_VpcConfig_SecurityGroupIds (940.63s)
=== CONT  TestAccAWSEksCluster_Version
--- PASS: TestAccAWSEksCluster_VpcConfig_EndpointPrivateAccess (2220.05s)
=== CONT  TestAccAWSEksCluster_EncryptionConfig
--- PASS: TestAccAWSEksCluster_VpcConfig_EndpointPublicAccess (1669.62s)
--- PASS: TestAccAWSEksCluster_Tags (905.21s)
--- PASS: TestAccAWSEksCluster_EncryptionConfig (875.73s)
--- PASS: TestAccAWSEksCluster_Version (2711.32s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	4706.341s
```

etoews commented:
@anGie44 We're in need of this functionality too. The PR is looking good so far. Does anything else need to happen before it can be merged and released? Thanks.

anGie44 commented on aws/resource_aws_eks_cluster_test.go:
```diff
@@ -424,6 +424,43 @@ func TestAccAWSEksCluster_VpcConfig_PublicAccessCidrs(t *testing.T) {
 	})
 }
 
+func TestAccAWSEksCluster_NetworkConfig_ServiceIpv4Cidr(t *testing.T) {
```
if you don't mind, can you add a test for the case where the empty config block (`kubernetes_network_config {}`) is provided to test the case where AWS supplies the IP?

philnichol replied:
Good shout, have added this to the _Basic test

anGie44 commented on aws/resource_aws_eks_cluster_test.go:
```diff
@@ -779,3 +816,24 @@ resource "aws_eks_cluster" "test" {
 }
+
+func testAccAWSEksClusterConfig_NetworkConfig_ServiceIpv4Cidr(rName string, serviceIpv4Cidr string) string {
+	return fmt.Sprintf(`
+%[1]s
+
+resource "aws_eks_cluster" "test" {
+  name     = %[2]q
+  role_arn = aws_iam_role.test.arn
+
+  vpc_config {
+    subnet_ids = aws_subnet.test[*].id
+  }
+
+  kubernetes_network_config {
+    service_ipv4_cidr = %s
+  }
+
+  depends_on = [aws_iam_role_policy_attachment.test-AmazonEKSClusterPolicy]
+}
+`, testAccAWSEksClusterConfig_Base(rName), rName, serviceIpv4Cidr)
+}
```
looks great overall, just a small nitpick to use the helper func `composeConfig` to concatenate the base config here 
```suggestion
	return composeConfig(testAccAWSEksClusterConfig_Base(rName), fmt.Sprintf(`
resource "aws_eks_cluster" "test" {
  name     = %q
  role_arn = aws_iam_role.test.arn

  vpc_config {
    subnet_ids = aws_subnet.test[*].id
  }

  kubernetes_network_config {
    service_ipv4_cidr = %s
  }

  depends_on = [aws_iam_role_policy_attachment.test-AmazonEKSClusterPolicy]
}
`, rName, serviceIpv4Cidr)
}
```

anGie44 commented on aws/resource_aws_eks_cluster_test.go:
```diff
@@ -424,6 +424,43 @@ func TestAccAWSEksCluster_VpcConfig_PublicAccessCidrs(t *testing.T) {
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccAWSEksClusterConfig_NetworkConfig_ServiceIpv4Cidr(rName, `"10.10.0.0/16"`),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSEksClusterExists(resourceName, &cluster2),
+					testAccCheckAWSEksClusterNotRecreated(&cluster1, &cluster2),
+					resource.TestCheckResourceAttr(resourceName, "kubernetes_network_config.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "kubernetes_network_config.0.service_ipv4_cidr", "10.10.0.0/16"),
+				),
+			},
```
since we want to ensure no further Op is needed when the IP is unchanged, we can use a `PlanOnly` check in place here
```suggestion
			{
				Config:   testAccAWSEksClusterConfig_NetworkConfig_ServiceIpv4Cidr(rName, `"10.10.0.0/16"`),
				PlanOnly: true,
				ExpectNonEmptyPlan: false,
			},
```

anGie44 replied:
it would also be great to add a test to verify that the resource is recreated in cases where the IP changes if time permits :)

anGie44 reviewed (changes requested):
Hi @philnichol, thank you for this PR! Off to a really great start 👍 A couple comments and suggestions to help address the current acctest failure and request for test coverage. If you have any questions or concerns, please reach out!

anGie44 commented on aws/resource_aws_eks_cluster.go:
```diff
@@ -334,6 +359,12 @@ func resourceAwsEksClusterRead(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("error setting vpc_config: %s", err)
 	}
 
+	if _, ok := d.GetOk("kubernetes_network_config"); ok {
+		if err := d.Set("kubernetes_network_config", flattenEksNetworkConfig(cluster.KubernetesNetworkConfig)); err != nil {
+			return fmt.Errorf("error setting network config: %s", err)
+		}
+	}
```
test output atm:
```
 TestAccAWSEksCluster_NetworkConfig_ServiceIpv4Cidr: resource_aws_eks_cluster_test.go:433: ImportStateVerify attributes not equivalent. Difference is shown below. Top is actual, bottom is expected.
        
        (map[string]string) {
        }
        
        
        (map[string]string) (len=3) {
         (string) (len=27) "kubernetes_network_config.#": (string) (len=1) "1",
         (string) (len=29) "kubernetes_network_config.0.%": (string) (len=1) "1",
```

philnichol commented:
@anGie44  thanks for the feedback! All great points! I think I've addressed your comments but let me know if I've missed anything.
Here's the updated output from the acceptance test:
```
$ TF_ACC=1 go test ./aws -v -count 1 -parallel 4 -run=TestAccAWSEksCluster_ -timeout 120m
=== RUN   TestAccAWSEksCluster_basic
=== PAUSE TestAccAWSEksCluster_basic
=== RUN   TestAccAWSEksCluster_EncryptionConfig
=== PAUSE TestAccAWSEksCluster_EncryptionConfig
=== RUN   TestAccAWSEksCluster_Version
=== PAUSE TestAccAWSEksCluster_Version
=== RUN   TestAccAWSEksCluster_Logging
=== PAUSE TestAccAWSEksCluster_Logging
=== RUN   TestAccAWSEksCluster_Tags
=== PAUSE TestAccAWSEksCluster_Tags
=== RUN   TestAccAWSEksCluster_VpcConfig_SecurityGroupIds
=== PAUSE TestAccAWSEksCluster_VpcConfig_SecurityGroupIds
=== RUN   TestAccAWSEksCluster_VpcConfig_EndpointPrivateAccess
=== PAUSE TestAccAWSEksCluster_VpcConfig_EndpointPrivateAccess
=== RUN   TestAccAWSEksCluster_VpcConfig_EndpointPublicAccess
=== PAUSE TestAccAWSEksCluster_VpcConfig_EndpointPublicAccess
=== RUN   TestAccAWSEksCluster_VpcConfig_PublicAccessCidrs
=== PAUSE TestAccAWSEksCluster_VpcConfig_PublicAccessCidrs
=== RUN   TestAccAWSEksCluster_NetworkConfig_ServiceIpv4Cidr
=== PAUSE TestAccAWSEksCluster_NetworkConfig_ServiceIpv4Cidr
=== CONT  TestAccAWSEksCluster_basic
=== CONT  TestAccAWSEksCluster_VpcConfig_EndpointPrivateAccess
=== CONT  TestAccAWSEksCluster_NetworkConfig_ServiceIpv4Cidr
=== CONT  TestAccAWSEksCluster_VpcConfig_PublicAccessCidrs
--- PASS: TestAccAWSEksCluster_basic (810.21s)
=== CONT  TestAccAWSEksCluster_Logging
--- PASS: TestAccAWSEksCluster_VpcConfig_PublicAccessCidrs (1486.56s)
=== CONT  TestAccAWSEksCluster_VpcConfig_SecurityGroupIds
--- PASS: TestAccAWSEksCluster_Logging (1257.63s)
=== CONT  TestAccAWSEksCluster_VpcConfig_EndpointPublicAccess
--- PASS: TestAccAWSEksCluster_NetworkConfig_ServiceIpv4Cidr (2302.95s)
=== CONT  TestAccAWSEksCluster_Tags
--- PASS: TestAccAWSEksCluster_VpcConfig_SecurityGroupIds (900.22s)
=== CONT  TestAccAWSEksCluster_Version
--- PASS: TestAccAWSEksCluster_VpcConfig_EndpointPrivateAccess (2718.91s)
=== CONT  TestAccAWSEksCluster_EncryptionConfig
--- PASS: TestAccAWSEksCluster_Tags (934.24s)
--- PASS: TestAccAWSEksCluster_EncryptionConfig (737.58s)
--- PASS: TestAccAWSEksCluster_VpcConfig_EndpointPublicAccess (1723.76s)
--- PASS: TestAccAWSEksCluster_Version (2832.07s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       5218.886s
```

etoews commented:
@anGie44 Thanks a lot for that initial review. Everything is looking good here. 

Do you have some time to give this another pass?

jasonyihk commented:
we also in need for this functionality, please release asap. tks

shrutilamba commented:
We need this functionality as well. How long will this take to be released?

anGie44 reviewed (changes requested):
Hi @philnichol (and those following this feature 😄 )-- apologies for the delay here ! a couple final notes before we can merge this in 👍

anGie44 commented on website/docs/r/eks_cluster.html.markdown:
```diff
@@ -156,6 +156,7 @@ The following arguments are supported:
 * `vpc_config` - (Required) Nested argument for the VPC associated with your cluster. Amazon EKS VPC resources have specific requirements to work properly with Kubernetes. For more information, see [Cluster VPC Considerations](https://docs.aws.amazon.com/eks/latest/userguide/network_reqs.html) and [Cluster Security Group Considerations](https://docs.aws.amazon.com/eks/latest/userguide/sec-group-r ...
 * `enabled_cluster_log_types` - (Optional) A list of the desired control plane logging to enable. For more information, see [Amazon EKS Control Plane Logging](https://docs.aws.amazon.com/eks/latest/userguide/control-plane-logs.html)
 * `encryption_config` - (Optional) Configuration block with encryption configuration for the cluster. Only available on Kubernetes 1.13 and above clusters created after March 6, 2020. Detailed below.
+* `kubernetes_network_config` - (Optional) Configuration block with kubernetes network configuration for the cluster. Detailed below.
```
Let's also a note here that if this block is removed from the configuration though, its essentially a no-op in the prior state so we can't do drift detection unless it's configured

philnichol replied:
good shout, have done this, let me know if the phrasing is alright.

philnichol commented on aws/resource_aws_eks_cluster_test.go:
```diff
@@ -491,6 +555,16 @@ func testAccCheckAWSEksClusterDestroy(s *terraform.State) error {
+func testAccCheckAWSEksClusterRecreated(i, j *eks.Cluster) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if aws.TimeValue(i.CreatedAt) == aws.TimeValue(j.CreatedAt) {
+			return errors.New("EKS Cluster was recreated")
```
Good spot!

philnichol commented on aws/resource_aws_eks_cluster.go:
```diff
@@ -113,6 +114,28 @@ func resourceAwsEksCluster() *schema.Resource {
 				},
 			},
+
+			"kubernetes_network_config": {
```
Hi @anGie44 , no worries at all. Happy to have a crack at the data_source, but if it takes a while I might well ask for your help since I don't want to hold people up :smile:. Before making any changes I seem to be getting the following error on all datasource tests, but can't see what I'm doing wrong, any ideas? I get the same (resourceNotFound) regardless of which datasource test I run it seems.
```
TF_ACC=1 go test ./aws -v -count 1 -parallel 10 -run=TestAccAWSEksClusterDataSource_basic
=== RUN   TestAccAWSEksClusterDataSource_basic
=== PAUSE TestAccAWSEksClusterDataSource_basic
=== CONT  TestAccAWSEksClusterDataSource_basic
    data_source_aws_eks_cluster_test.go:18: Step 1/1 error: terraform failed: exit status 1

        stderr:

        Error: error reading EKS Cluster (tf-acc-test-gs09q): ResourceNotFoundException: No cluster found for name: tf-acc-test-gs09q.
        {
          RespMetadata: {
            StatusCode: 404,
            RequestID: "91f46140-c965-4387-bddf-9834ad7beb69"
          },
          Message_: "No cluster found for name: tf-acc-test-gs09q."
        }


--- FAIL: TestAccAWSEksClusterDataSource_basic (5.87s)
FAIL
FAIL    github.com/terraform-providers/terraform-provider-aws/aws       5.909s
FAIL
```

anGie44 replied:
Ahh which version of Terraform do you have locally? I've come across these type of errors when using the family of `0.13` versions. I'd recommend`v.0.12.29` or `0.14+`

philnichol replied:
🤦 sorry I didn't try that before, works now, will aim to address this and the documentation comment shortly, thanks @anGie44 !

anGie44 replied:
no worries - i don't think we have that behavior documented unfortunately ..sounds great 👍

philnichol replied:
@anGie44 I've pushed my changes to the data source, let me know if there are any issues. Acceptance tests ran here:

```
$ TF_ACC=1 go test ./aws -v -count 1 -parallel 10 -run=TestAccAWSEksClusterDataSource_basic -timeout 60m
=== RUN   TestAccAWSEksClusterDataSource_basic
=== PAUSE TestAccAWSEksClusterDataSource_basic
=== CONT  TestAccAWSEksClusterDataSource_basic
--- PASS: TestAccAWSEksClusterDataSource_basic (998.81s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       998.846s
```

anGie44 replied:
thanks @philnichol -- test looks good on my end as well 👍 i'm running the resource acceptance tests as well in the background (I don't expect any changes)..and will comment back w/the PR approval

etoews commented on aws/resource_aws_eks_cluster.go:
```diff
@@ -113,6 +114,28 @@ func resourceAwsEksCluster() *schema.Resource {
+							ForceNew: true,
+							ValidateFunc: validation.All(
+								validation.IsCIDRNetwork(12, 24),
+								validation.StringMatch(regexp.MustCompile(`^(10|172\.(1[6-9]|2[0-9]|3[0-1])|192\.168)\..*`), "must be within 10.0.0.0/8, 172.16.0.0.0/12, or 192.168.0.0/16"),
```
Please add the 100.64.0.0/10 range here too as per [How do I use multiple CIDR ranges with Amazon EKS?](https://aws.amazon.com/premiumsupport/knowledge-center/eks-multiple-cidr-ranges/) and [Amazon EKS now supports additional VPC CIDR blocks](https://aws.amazon.com/about-aws/whats-new/2018/10/amazon-eks-now-supports-additional-vpc-cidr-blocks/).

Sounds like the 198.19.0.0/16 range is valid in this scenario too but I'm not familiar with that and wouldn't want any ambiguity around it holding up this PR.

anGie44 replied:
thanks for bringing this up @etoews! looks like even though the doc suggests it for use in associating the range to a VPC, we're unfortunately blocked by the AWS API's RFC1918 validation e.g. i gave it a shot locally (removing the validation funcs in place for testing) and passing the CIDR `100.64.0.0/16` (since the netmask size must be between 12 and 24 inclusive):
```
 Error: error creating EKS Cluster (tf-acc-test-blljj): InvalidParameterException: Invalid Cluster Service Ipv4 CIDR for the cluster tf-acc-test-blljj, must be within RFC1918 range
        {
          RespMetadata: {
            StatusCode: 400,
            RequestID: "dcc64171-1ffa-4388-a1e1-e05f4d6cb1d0"
          },
          ClusterName: "tf-acc-test-blljj",
          Message_: "Invalid Cluster Service Ipv4 CIDR for the cluster tf-acc-test-blljj, must be within RFC1918 range"
        }
```

anGie44 replied:
and similar response when passing in `198.19.0.0/16`

etoews replied:
Yep. I was thinking about another use for those ranges. My misunderstanding. The 100.64.0.0/10 and 198.19.0.0/16 ranges are for pod IP ranges, whereas this PR is for service IP ranges. Apologies for the noise!

anGie44 replied:
no worries at all @etoews!

etoews commented on website/docs/r/eks_cluster.html.markdown:
```diff
@@ -180,6 +181,18 @@ The following arguments are supported in the `provider` configuration block:
+
+* `service_ipv4_cidr` - (Optional) The CIDR block to assign Kubernetes service IP addresses from. If you don't specify a block, Kubernetes assigns addresses from either the 10.100.0.0/16 or 172.20.0.0/16 CIDR blocks. We recommend that you specify a block that does not overlap with resources in other networks that are peered or connected to your VPC. You can only specify a custom CIDR block when y ...
+
+    * Within one of the following private IP address blocks: 10.0.0.0/8, 172.16.0.0.0/12, or 192.168.0.0/16.
```
You'd need to doc the 100.64.0.0/10 range (and 198.19.0.0/16 range if you want to add it) here too.

anGie44 reviewed (approved):
Thanks again for this feature @philnichol 🚀 
Output of acceptance tests:
```
--- PASS: TestAccAWSEksClusterAuthDataSource_basic (17.33s)
--- PASS: TestAccAWSEksCluster_EncryptionConfig (859.07s)
--- PASS: TestAccAWSEksCluster_Logging (895.26s)
--- PASS: TestAccAWSEksCluster_basic (919.82s)
--- PASS: TestAccAWSEksCluster_Tags (948.84s)
--- PASS: TestAccAWSEksClusterDataSource_basic (969.99s)
--- PASS: TestAccAWSEksCluster_VpcConfig_SecurityGroupIds (999.36s)
--- PASS: TestAccAWSEksCluster_VpcConfig_PublicAccessCidrs (1377.22s)
--- PASS: TestAccAWSEksCluster_VpcConfig_EndpointPublicAccess (1632.31s)
--- PASS: TestAccAWSEksCluster_NetworkConfig_ServiceIpv4Cidr (2127.25s)
--- PASS: TestAccAWSEksCluster_VpcConfig_EndpointPrivateAccess (2502.05s)
--- PASS: TestAccAWSEksCluster_Version (2609.50s)
```
